addons/binding/org.openhab.binding.neato/src/main/java/org/openhab/binding/neato/internal/NeatoRobot.java

```diff
@@ -0,0 +1,229 @@
+        String stringToSign = this.serialNumber.toLowerCase() + "\n" + dateString + "\n" + body;
+
+        SecretKeySpec secret_key = new SecretKeySpec(this.secret.getBytes("UTF-8"), "HmacSHA256");
+        sha256_HMAC.init(secret_key);
+
+        byte[] signature = sha256_HMAC.doFinal(stringToSign.getBytes("UTF-8"));
+        String hexString = Hex.encodeHexString(signature);
+
+        // Properties headers = new Properties
+        Properties headers = new Properties();
+        headers.setProperty("Date", dateString);
+        headers.setProperty("Authorization", "NEATOAPP " + hexString);
+        headers.setProperty("Accept", "application/vnd.neato.nucleo.v1");
+
+        InputStream stream = new ByteArrayInputStream(body.getBytes(StandardCharsets.UTF_8));
+
+        String result = HttpUtil.executeUrl("POST",
+                "https://nucleo.neatocloud.com:4443/vendors/neato/robots/" + this.serialNumber + "/messages", headers,
+                stream, "text/html; charset=ISO-8859-1", 20000);
+
+        return result;
+    }
+
+    public Boolean sendCommand(Command command) throws Exception {
+        String body = "";
+
+        if (command.toString().equalsIgnoreCase("clean")) {
+
+            if (this.state.getAvailableServices().getHouseCleaning().equalsIgnoreCase("basic-1")) {
+                body = "{\"reqId\": \"1\", \"cmd\": \"startCleaning\", \"params\": { \"category\": 2, \"mode\": 2, \"modifier\": 2}}";
+            } else if (this.state.getAvailableServices().getHouseCleaning().equalsIgnoreCase("minimal-2")) {
+                body = "{\"reqId\": \"1\", \"cmd\": \"startCleaning\", \"params\": { \"category\": 2, \"navigationMode\": 2}}";
+            } else if (this.state.getAvailableServices().getHouseCleaning().equalsIgnoreCase("basic-2")) {
+                body = "{\"reqId\": \"1\", \"cmd\": \"startCleaning\", \"params\": { \"category\": 2, \"mode\": 2, \"modifier\": 1, \"navigationMode\": 1}}";
+            } else {
+                logger.error("No matching service for houseCleaning. Will not start house cleaning!");
+            }
+        } else if (command.toString().equalsIgnoreCase("pause")) {
+            body = "{\"reqId\": \"1\", \"cmd\": \"pauseCleaning\"}";
+        } else if (command.toString().equalsIgnoreCase("stop")) {
+            body = "{\"reqId\": \"1\", \"cmd\": \"stopCleaning\"}";
+        } else if (command.toString().equalsIgnoreCase("resume")) {
+            body = "{\"reqId\": \"1\", \"cmd\": \"resumeCleaning\"}";
+        } else if (command.toString().equalsIgnoreCase("dock")) {
+            body = "{\"reqId\": \"1\", \"cmd\": \"sendToBase\"}";
+        } else if (command.toString().equalsIgnoreCase("dismissAlert")) {
+            body = "{\"reqId\": \"1\", \"cmd\": \"dismissCurrentAlert\"}";
+        }
+
+        if (body.isEmpty()) {
+            return false;
+        }
+
+        this.callNeatoWS(body);
+
+        return true;
+    }
+
+    public Boolean sendGetRobotInfo() throws Exception {
```

Comment:
Methods show never be declared to throw `Exception` which means expect the worst, I'll throw anything at you.